langwatch/src/components/analytics/reports/ReportGrid.tsx

```diff
@@ -0,0 +1,234 @@
+import { Box, Card, Grid, HStack, Text } from "@chakra-ui/react";
+import {
+  closestCenter,
+  DndContext,
+  type DragEndEvent,
+  type DragStartEvent,
+  DragOverlay,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import { rectSortingStrategy, SortableContext } from "@dnd-kit/sortable";
+import { useState } from "react";
+import { BarChart2 } from "lucide-react";
+import {
+  DraggableGraphCard,
+  type GraphData,
+  type SizeOption,
+} from "./DraggableGraphCard";
+
+interface ReportGridProps {
+  graphs: GraphData[];
+  projectSlug: string;
+  onGraphDelete: (graphId: string) => void;
+  onGraphSizeChange: (graphId: string, size: SizeOption) => void;
+  onGraphsReorder: (
+    layouts: Array<{
+      graphId: string;
+      gridColumn: number;
+      gridRow: number;
+      colSpan: number;
+      rowSpan: number;
+    }>,
+  ) => void;
+  deletingGraphId: string | null;
+}
+
+export function ReportGrid({
```

Comment:
Similar comment as above -- consider that this actually just a grid, and that you can use it as a ReportGrid in its implementation. This makes less work for the next engineer that might need to reuse this UI, which isn't actually coupled to the reports except in naming for now.